libavcodec/hevcdec.c

```diff
@@ -571,7 +574,9 @@ static int set_sps(HEVCContext *s, const HEVCSPS *sps,
     }
 
     s->ps.sps = sps;
-    s->ps.vps = s->ps.vps_list[s->ps.sps->vps_id];
+    if (s->ps.vps_list && s->ps.vps_list[s->ps.sps->vps_id]) {
+        s->ps.vps = (HEVCVPS*) s->ps.vps_list[s->ps.sps->vps_id];
```

Comment:
looks like you don't need the conversion?

Reply:
Done.